internal/pkg/cli/secret_init.go

```diff
@@ -354,13 +355,18 @@ func (o *secretInitOpts) askForSecretValues() error {
 func (o *secretInitOpts) RecommendedActions() {
 }
 
+type errPutSecretFailed interface {
```

Comment:
Hmm why do we create `errPutSecretFailed` instead of using `errSecretFailedInSomeEnvironments`

Reply:
We can I just thought it might not be extensible... although right now `errSecretFailedInSomeEnvironments` is indeed the only reason why a put secret can fail

Reply:
👍 agreed with 👴 @iamhopaul123 

Like you said interfaces provide us flexibility, but while we have only one type implementing the interface let's just the type directly. Once the need emerges, refactoring to add an interface becomes natural. Otherwise, we end up adding an additional type to our package namespace. All this is a nitpick tho since the interface is private in the pkg.